docs/glossary.rst

```diff
@@ -0,0 +1,297 @@
+
+        .. math::
+
+          \min_{P>0}\, \varepsilon \text{KL}(P|K)
+          + \rho_a \text{KL}(P\mathbf{1}_m | a)
+          + \rho_b \text{KL}(P^T \mathbf{1}_n | b)
+
+        The *original* OT problem taught in linear programming courses is
+        recovered by using the formulation above relying on the cost :math:`C`,
+        and letting :math:`\varepsilon \rightarrow 0`, and
+        :math:`\rho_a, \rho_b \rightarrow \infty`.
+        In that case the entropy disappears, whereas the :math:`\text{KL}`
+        regularization above become constraints on the marginals of :math:`P`:
+        This results in a standard min cost flow problem also called the
+        :term:`Kantorovich problem`.
+
+        The *balanced* regularized OT problem is recovered for finite
+        :math:`\varepsilon > 0` but letting :math:`\rho_a, \rho_b \rightarrow
+        \infty`. This problem can be shown to be equivalent to a matrix scaling
+        problem, which can be solved using the Sinkhorn fixed-point algorithm.
+        To handle the case :math:`\rho_a, \rho_b \rightarrow \infty`, the
+        Sinkhorn function uses parameters ``tau_a`` and ``tau_b`` equal
+        respectively to :math:`\rho_a /(\varepsilon + \rho_a)` and
+        :math:`\rho_b / (\varepsilon + \rho_b)` instead. Setting either of these
+        parameters to 1 corresponds to setting the corresponding
+        :math:`\rho_a, \rho_b` to :math:`\infty`.
+
+    envelope theorem
+        The envelope theorem is a major result about the differentiability
+        properties of the value function of a parameterized optimization
+        problem. Namely, that for a function :math:`f` defined implicitly as an
+        optimal objective parameterized by a vector :math:`x`,
+
+        .. math::
+          h(x):=\min_z s(x,z), z^\star(x):=\arg\min_z s(x,z)
+
+        one has
+
+        .. math::
+          \nabla h(x)=\nabla_1 s(x,z^\star(x))
+
+        stating in effect that the optimal :math:`z^\star(x)` does not
+        need to be differentiated w.r.t. :math:`x` when computing the
+        gradient of :math:`h`. Note that this result is not valid for higher
+        order differentiation.
+
+    ground cost
+        A real-valued function of two variables, :math:`c(x,y)` that describes
+        the cost needed to displace a point :math:`x` in a source measure to
+        :math:`y` in a target measure.
+
+    implicit differentiation
+        Differentiation technique to compute the vector-Jacobian
+        product of the minimizer of an optimization procedure by considering
+        that small variations in the input would still result in minimizers
+        that verify optimality conditions (KKT or first-order conditions). These
+        identities can then help recover the vector-Jacobian operator by
+        inverting a linear system.
+
+    input-convex neural networks
+        A neural network architecture for vectors with a few distinguishing
```

Comment:
Maybe add some citations here, e.g., Makkuva and Amos?

Reply:
I think that if we start adding citations in a glossary, we will then need to add literally hundreds of them to be fair :) I think that a glossary is usually citation free (since the goal is to describe things in simple terms)

Reply:
Ok, fair, let's not add them in the glossary.